addon/components/base-layer.js

```diff
@@ -139,20 +139,23 @@ export default Ember.Component.extend(
       @method search
       @param {Object} e Event object.
       @param {<a href="http://leafletjs.com/reference-1.0.0.html#latlng">L.LatLng</a>} e.latlng Center of the search area.
-      @param {Object[]} layer Object describing layer that must be searched.
+      @param {Object[]} layerModel Object describing layer that must be searched.
       @param {Object} searchOptions Search options related to layer type.
       @param {Object} results Hash containing search results.
       @param {Object[]} results.features Array containing (GeoJSON feature-objects)[http://geojson.org/geojson-spec.html#feature-objects]
       or a promise returning such array.
     */
     _search(e) {
-      let shouldSearch = e.layer === this.get('layer');
+      let shouldSearch = typeof (e.filter) === 'function' && e.filter(this);
```

Comment:
Send outside a reference to a component instance (in e.filter callback), is bad practice.
Will be better to send JSON-object containing only filter-needed component's properties.

Also if search logic is refactored, then identification logic must be refactored  like that too.